Editable ascii

firexx opened the pull request.

Simsys commented:
Some additional remarks:

There are some changes in this pull request, which are not necessary. There is no need to change the interface to the outside world (qhexedit.h), to introduce the possibility to use the ascii area for input. There are some people using QHexEdit as a library and if we change the interface, they have to change their code. They will not like to do this. This is bad for the acceptance of QHexEdit.

We need no class CursorPosition. We can use a QHexEdit private variable _editAreaIsAscii and handle all inside QHexEdit Class. The cursorposition should remain qint64. The _editAreaIsAscii  value is not interesting for the outside world, I think.

You created an extra method keyPressedEventAscii(). In this method there is some code parts identical to the original keyPressedEvent(). We do not want to double source code. That is bad to maintain. Please do not double code parts.

Thanks again for Your work. I know, that it is not easy to accept my demands, because You invested a lot of time into QHexEdit. But I want to bind things together.

I would be very happy, if You could submit a pull request with no conflicts, with the above mentioned changed and request, which addresses only the feature asciiArea edit (and nothing else).

Simsys commented:
Please first rebase to v0.8.2, then input/merge the changes.

Thank You so much

firexx commented:
Hi Simsys,

it is a good idea to use some private variable instead of specific class.
if i got more time i'll re-implement this part.

the extra method was simpler. in the first redaction it was completely
implemented in the keypressEvent. with 1000 if's by calculating and
reactions. then i throws all of them away and implemented in separated
function again. i also dislike to use the double code, but in this case it
was a better way.

the mostly problem is to delete a multiplies "* 2" may be i can at the
begin of the function set the coefficient to 1 or 2 depends of the edit
region and use coefficient in code instead of constants. if i throws
CursorPosition the other problems be solved automatically.






On Tue, Dec 20, 2016 at 12:26 PM, Simsys <notifications@github.com> wrote:

> Please first rebase to v0.8.2, then input/merge the changes.
>
> Thank You so much
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Simsys/qhexedit2/pull/51#issuecomment-268220144>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/APXyu5gA40H8uAfWv9j3PKdmf_02lzx_ks5rJ7tggaJpZM4LRrbs>
> .
>



-- 
Mit freundlichen Grüßen

Andrej Kostrov

firexx commented:
make it again better
